### ISSUE TITLE
Custom Objective's not creating HQ's

### ISSUE THREAD

**Issue by Burdy323** [ALiVEOS/ALiVE.OS#381]
As the title says : https://ufile.io/dezwy 

Note how the logistic module does not pick up the objective as a resupply point & how no object is placed to represent the HQ

**Comment by tupolov** [comment]
Please provide:

1. Unbinarized SQM File
2. RPT
3. Description of how you have configured the relevant modules
4. Description of how to repro the issue
5. Screenshots showing the module params

**Comment by Burdy323** [comment]
1. https://ufile.io/oa75d
2. https://ufile.io/0us5h
3/5. https://pasteboard.co/GznTH4pY.png
4. Place down a Military Commander, Logistics, ALIVE required, AI profile module, Custom Objective, & a TAOR marker. Make sure the TAOR marker is within an area without any military installations. Note that ALIVE will not place down any logistics nor HQ points. Even when the TAOR marker is larger and encompasses multiple military installations, the custom objectives will never get read as  HQ/logistic points

**Comment by tupolov** [comment]
@Burdy323 Thanks for the info.

So the wiki entry is slightly incorrect (http://alivemod.com/wiki/index.php/Military_Placement_(Custom_Objectives))

CreateHQ no longer automatically spawns a composition, it looks for military buildings that are within the "size" of the objective (it does not use a TAOR). If it doesn't find any mil buildings it will not create an HQ.

Objective size dictates the search area for military buildings, the center point being the module position and the search radius being the size. Looks like you have a good size objective already, just ensure the module is in central position to cover the area you want it to.

So, you need to either manually place down a composition or have ALiVE spawn a composition (enter the config classname in the module parameter) - or keep increaseing the objective size.

**Comment by Burdy323** [comment]
Alright thank you. So a question to you would be, is there anyway to get the module to pick up manually placed objects for creating the HQ?

I tried placing down some ammo boxes (box_IND_Ammo_F), followed by 
`[ALiVE_compositions,"FieldHQ",["box_IND_Ammo_F"]] call ALiVE_fnc_HashSet;`
in the module's init. However, it appears that nothing has changed and the module does not pick up the objects.

**Comment by tupolov** [comment]
Again, the wiki is slightly out of date :)

I recommend placing an ALiVE composition manually - (in 3DEN, go to Groups, select Empty and then choose Military > Field HQ) or specifying a composition classname in the module parameter Composition.

I'll clean up the wiki.

We are considering auto spawning an HQ composition if no HQ buildings are found and create HQ is selected - would this work for you?

**Comment by Burdy323** [comment]
Yeah that's what I ended up doing - cleaning up the wiki would be nice as it saves a lot of headaches!

To the auto spawning HQ - that seems like it would be the best plan of action as currently the custom objective feels a little vague in terms of what needs to be done (as the create an HQ sorta implies that an HQ will be created by the module automatically).

Perhaps a feature request - could it be possible to do what I was trying to do above and simply define an object for the module to pick up as the HQ? The issue arises with having to use zeus compositions as say you were to play a mod in another era - you have a modern day structures sticking out like a soar thumb.

**Comment by tupolov** [comment]
ALiVE is designed to handle "other era" maps, factions etc. For example we have indexed IFA maps and ALiVE is aware of WWII military style buildlings - if you placed one of those it would be used by this module. 

However, we don't have any WWII or Vietnam era compositions yet though, so you would need to manually place an HQ/building object.

**Comment by Burdy323** [comment]
That's what I am saying - is there anyway to get the custom objective module to pickup manually placed HQ objects? Or do I have to stick with compositions?

**Comment by tupolov** [comment]
Yes it will pick up manually placed objects such as Cargo Towers, barracks etc. The list of buildings it looks for are defined as part of the map index. Specifically these: 

```
ALIVE_militaryBuildingTypes
ALIVE_militaryHQBuildingTypes
```

I thought you were asking for it to pick up ANY object as the HQ object.

**Comment by Burdy323** [comment]
Alright thank you that clears it up. So essentially, if I manually place down any object that is seen as a military building from the index, the Custom Objective will detect that?

**Comment by tupolov** [comment]
Yep!

I'm going to add code so that if create HQ is selected and it doesn't find any HQ buildings it will spawn a field HQ composition automatically. Should clear up any confusion :)

**Comment by Burdy323** [comment]
Awesome! Also, i'd have to ask you to add code like what I was trying to use before ([ALiVE_compositions,"FieldHQ",["box_IND_Ammo_F"]] call ALiVE_fnc_HashSet;) - where in the init of the module you can define an object that is not apart of the index, yet still have it be used as an HQ. The reason being, that some maps (custom in particular) do not have many military objects. This leads to a situation in which the mission maker does not have many options when it comes to placing objects down which they may want to be used HQ's (as the index never defined more than a handful).

Also, final question - it appears that the HQ is being placed! However, despite the objective priority being 999, the logistics module is not detecting the HQ as a logistics hub and as a result, no units are spawning. Reason?

**Comment by tupolov** [comment]
Logistics doesn't spawn units. What are you expecting from Military Logistics?

**Comment by Burdy323** [comment]
Right - I am expecting that it will replenish the casualties and spawn them at the HQ. Let me rephrase this - I have the Military Placement module down on the map and set for 200 men. I then have the custom HQ within the TAOR. In theory this should allow the units to spawn at the HQ correct? I am however, having no units spawn in, despite the debugger showing that there is an HQ on the map and the military objective is placed with priority set at 999

**Comment by tupolov** [comment]
Your Mil Placement (Custom Objective) should place units, if you have specified a number of groups in the module parameters. Mil Logistics is only going to do something once they start getting killed :)

**Comment by Burdy323** [comment]
So does that mean that the mil placement module does not place units at the user defined HQ? 

I tried setting the modules settings to 0 for everything as I want the placement module to populate it, and it appears nothing will spawn that way - only when I manually place a number do units start spawning in

**Comment by tupolov** [comment]
Mil Placement (Mil Obj) will place units at any military areas within the TAOR. Those units will be tasked by OPCOM to attack/defend/reserve high priority objectives (Assuming there are any military objectives within your TAOR).

Mil Placement (Cust. Obj) will place units at the custom objective. As above OPCOM will take control of those forces and assign them to objectives.

You need either number of groups specified in the Custom Obj module or Force Size set in the Mil Placement module.

**Comment by Burdy323** [comment]
I am getting "warning no locations found" from my Mil Placement module, despite a 999 priority HQ enabled custom obj within the TAOR

**Comment by tupolov** [comment]
Mil Placement (Mil Obj) has nothing to do with Mil Placement (Cust Obj). It knows nothing about the custom objective. 

The first defines objectives and places units at military areas defined within the map.

The second defines an objective and places units at editor placed location/compositions.

Only the Military AI Commander will know about all the objectives.

Attached the mission you are trying to achieve.

[burdy.Altis.zip](https://github.com/ALiVEOS/ALiVE.OS/files/1122864/burdy.Altis.zip)

**Comment by Burdy323** [comment]
Yeah, what you sent is exactly the "issue" I am referring too. 

Essentially, I am making a few missions on smaller custom maps. I want all of the reinforcements/units to spawn at landing zones sprinkled throughout the map to simulate populated LZ's.

The "issue" (if you want to call it that), is that the maps are sparse on mil objects. Meaning that I was hoping I could place HQ's at each of the LZ's to act as placement/logistical replenishing areas, instead of having to rely on the single Military area on the map

**Comment by Burdy323** [comment]
The issue becomes exaggerated on smaller maps where you try to make push scenarios, as you essentially have nowhere for the logistics to come in as the Placement module ignores the custom objectives (which in turn leaves you with nowhere for reinforcements to flow in)

**Comment by tupolov** [comment]
You are still confused :)

The custom objective is a placement module, there is no ignoring going on.

You have 3 ways of placing units on the map.

1. Mil Placement (MILITARY) - define objectives and place units at military locations
2. Mil Placement (CIVILIAN) - define objectives and place units at civilian locations
3. Mil Placement (CUSTOM) - define an objective and place units at this objective.

OPCOM then takes control of all units placed and prioritizes objectives based on their priority setting. HQ is highest, so it will garrison and defend any HQ locations.

LOGCOM is another mechanic we will get to... but please understand the above first :)

**Comment by Burdy323** [comment]
So essentially if I were to achieve what I wanted to do : ignore placing down the Military Placement module, use the Custom Obj module's count function to populate the map, and then anything that is killed off will be reinforced via LOGCOM at any of those custom objectives which have HQ's?

**Comment by tupolov** [comment]
Awesome, yes.

LOGCOM has two modes - Dynamic or Fixed, play around with both to see what works for your scenario. If its a push scenario then Dynamic might fit better.

Didn't mean to sound condescending earlier with the confused comment.

Let us know how you get on and thanks for using ALiVE!

**Comment by Burdy323** [comment]
It's no worries, sometimes the wiki can be misleading/vague.

Stay tuned for another "bug" report - having a hard time getting custom planes working with the new MACC module! LOL

**Comment by tupolov** [comment]
Yes apologies, wiki updated!

Did you see the MACC tutorial here - http://alivemod.com/forum/3042-alive-military-air-component-commander-tutorial

**Comment by Burdy323** [comment]
Yeah I did - the issue I am having is that ALIVE does not (I assume) recognize the types of planes for the mod I am using (UNSUNG) - essentially, my manually placed aircraft will simply sit on the carrier with repeated calls for CAP followed by "aircraft not available". Pilots spawn but no one boards their aircraft. I tried it with both the default UNSUNG_W as well as making my own faction which included just the aircraft I was using to no avail. However, as soon as I dropped down a default Blu_F aircraft, the module worked like a charm.

**Comment by tupolov** [comment]
Can you create a ticket for this issue?

**Comment by Burdy323** [comment]
Will do

**Comment by tupolov** [comment]
Fixed with https://github.com/ALiVEOS/ALiVE.OS/commit/039a0bc16020f899a5c08420916407686a9c82b0
